staging/src/k8s.io/client-go/tools/cache/delta_fifo.go

```diff
@@ -407,6 +420,24 @@ func (f *DeltaFIFO) Pop(process PopProcessFunc) (interface{}, error) {
 			f.cond.Wait()
 		}
 		id := f.queue[0]
+		if id == resyncEndKey || id == relistEndKey {
```

Comment:
I'm suspicious of markers.  Are there comments in here explaining why this was the only solution?